### PR TITLE
Add DynamicallyAccessedMemberTypes.Interfaces enum value

### PR DESCRIPTION
**Pull request by vitek-karas**
Implements new API approved in https://github.com/dotnet/runtime/issues/51487.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/DynamicallyAccessedMemberTypes.cs [hunk 1]
```diff
@@ -87,6 +87,11 @@ enum DynamicallyAccessedMemberTypes
         /// </summary>
         NonPublicEvents = 0x1000,
 
+        /// <summary>
+        /// Specifies all interfaces implementations
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/DynamicallyAccessedMemberTypes.cs:91]: Is `implementations` the right word to use here? I immediately thought this means the members that implement an interface, but not necessarily the interfaces themselves. The intention is that this allows for calling `.GetInterfaces()`. How about

```suggestion
        /// Specifies all interfaces implemented by the type.
```

(nit) the others all have a `.` at the end.

**Review by eerhardt — APPROVED** [review]
LGTM.

Do we have an issue for marking `.GetInterfaces()` and friends with this new value?

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 2]
```diff
@@ -6082,6 +6082,7 @@ public enum DynamicallyAccessedMemberTypes
         NonPublicProperties = 1024,
         PublicEvents = 2048,
         NonPublicEvents = 4096,
+        Interfaces = 8192,
```
Comment (stephentoub) [inline src/libraries/System.Runtime/ref/System.Runtime.cs:6085]: Is there anywhere we should be using this now that we have it?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Runtime/ref/System.Runtime.cs (hunk 2 above)
Comment (eerhardt) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:6085]: Yes. But that will require a few changes throughout the repo, and we will need the linker to start respecting this new value. I assume this is to unblock the linker work.

Comment (vitek-karas) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:6085]: Exactly - we need to flow this to linker, teach linker about it and then we can apply the runtime/framework changes.

Comment (vitek-karas) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:6085]: Basically tracked by https://github.com/mono/linker/issues/1731.

**Comment by vitek-karas** [comment]
There's no runtime issue for annotating GetInterfaces - but there's basically a linker counterpart here https://github.com/mono/linker/issues/1731 which will require it.

**Pull request merged**